pkg/scaffold/olm-catalog/config.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package catalog
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-sdk/pkg/scaffold"
+
+	"github.com/ghodss/yaml"
+)
+
+// CSVConfig is a configuration file for CSV composition. Its fields contain
+// file path information.
+type CSVConfig struct {
+	// The operator manifest file path. Defaults to deploy/operator.yaml.
+	OperatorPath string `json:"operator-path,omitempty"`
+	// A list of CRD and CR manifest file paths. Defaults to deploy/crds.
+	CRDCRPaths []string `json:"crd-cr-paths,omitempty"`
+	// The RBAC role manifest file path. Defaults to deploy/role.yaml.
+	RolePath string `json:"role-path,omitempty"`
+}
+
+func getCSVConfig(cfgFile string) (*CSVConfig, error) {
+	cfg := &CSVConfig{}
+	if _, err := os.Stat(cfgFile); err == nil {
+		cfgData, err := ioutil.ReadFile(cfgFile)
+		if err != nil {
+			return nil, err
+		}
+		if err = yaml.Unmarshal(cfgData, cfg); err != nil {
+			return nil, err
+		}
+	} else if !os.IsNotExist(err) {
+		return nil, err
+	}
+
+	if err := cfg.setFields(); err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
+const yamlExt = ".yaml"
+
+func (c *CSVConfig) setFields() error {
+	if c.OperatorPath == "" {
+		info, err := (&scaffold.Operator{}).GetInput()
+		if err != nil {
+			return err
+		}
+		c.OperatorPath = info.Path
+	}
+
+	if c.RolePath == "" {
+		info, err := (&scaffold.Role{}).GetInput()
+		if err != nil {
+			return err
+		}
+		c.RolePath = info.Path
+	}
+
+	if len(c.CRDCRPaths) == 0 {
+		paths, err := getManifestPathsFromDir(scaffold.CrdsDir)
+		if err != nil {
+			return err
+		}
+		c.CRDCRPaths = paths
+	} else {
+		// Allow user to specify a list of dirs to search. Avoid duplicate files.
+		paths, seen := make([]string, 0), make(map[string]struct{})
+		for _, path := range c.CRDCRPaths {
+			finfo, err := os.Stat(path)
```

Comment:
Nit: Use camel case for two lettered names `fileInfo` or better yet, keep it short `info`.
E.g https://github.com/golang/go/blob/go1.11.4/src/path/filepath/path.go#L351
https://github.com/golang/go/wiki/CodeReviewComments#variable-names